code/game/objects/items/melee/misc.dm

```diff
@@ -74,10 +74,54 @@
 	custom_materials = list(/datum/material/iron = 1000)
 	wound_bonus = 10
 	bare_wound_bonus = 25
+	//lunge vars
+	COOLDOWN_DECLARE(lunge_cooldown)
+	var/lunge_cooldown_time = 5 SECONDS
+	var/is_lunging = FALSE
 
 /obj/item/melee/sabre/Initialize()
 	. = ..()
 	AddComponent(/datum/component/butchering, 30, 95, 5) //fast and effective, but as a sword, it might damage the results.
+	RegisterSignal(src, COMSIG_MOVABLE_IMPACT, .proc/lunge_hit)
+
+/obj/item/melee/sabre/examine(datum/source, mob/user, list/examine_list)
+	. = ..()
+	. += "<span class='notice'><b>right-click</b> to lunge with [src]. 5 second cooldown.</span>"
+
+/obj/item/melee/sabre/pickup(mob/user)
+	. = ..()
+	RegisterSignal(user, COMSIG_MOVABLE_IMPACT, .proc/lunge_hit)
+
+/obj/item/melee/sabre/dropped(mob/user, silent)
+	. = ..()
+	UnregisterSignal(src, COMSIG_MOVABLE_IMPACT)
+
+/obj/item/melee/sabre/attack_alt(mob/living/victim, mob/living/user, params)
+	return ALT_ATTACK_CANCEL_ATTACK_CHAIN //so trying to lunge doesn't also count as a normal attack (read: double dipping)
+
+/obj/item/melee/sabre/afterattack(atom/target, mob/user, proximity_flag, click_parameters)
+	. = ..()
+	var/list/params = params2list(click_parameters)
+
+	if(params["right"] && get_dist(user, target) < 3)
+		if(!COOLDOWN_FINISHED(src, lunge_cooldown))
+			var/time_left = COOLDOWN_TIMELEFT(src, lunge_cooldown)
+			to_chat(user, "<span class='warning'>You need to wait [DisplayTimeText(time_left)] before lunging again!</span>")
+			return
+		COOLDOWN_START(src, lunge_cooldown, lunge_cooldown_time)
+		is_lunging = TRUE
+		to_chat(user, "<span class='notice'>You lunge at [target]!</span>")
+
+		user.throw_at(target, 2, 10, spin = FALSE, gentle = TRUE)
+
+/obj/item/melee/sabre/proc/lunge_hit(mob/living/carbon/user, atom/hit)
+	SIGNAL_HANDLER
+
+	if(!is_lunging)
+		return
+	is_lunging = FALSE
```

Comment:
Instead of having `is_lunging`, register to the signal when it starts to lunge and unregister when it hits.